ddlite.py

```diff
@@ -227,6 +196,101 @@ def get_rule_priority_vote_accuracy(self, ground_truth, holdout_only=False):
           correct += 1 if self.rules[i,j] == gt[j] else 0
           break
     return float(correct) / len(gt)
+  
+  def __repr__(self):
+    return '\n'.join(str(e) for e in self.extractions)
+          
+class Relation:
+  def __init__(self, e1_idxs, e2_idxs, e1_label, e2_label, sent, xt):
+    self.e1_idxs = e1_idxs
+    self.e2_idxs = e2_idxs
+    self.idxs = [self.e1_idxs, self.e2_idxs]
+    self.e1_label = e1_label
+    self.e2_label = e2_label
+    self.labels = [self.e1_label, self.e2_label]
+
+    # Absorb XMLTree and Sentence object attributes for access by rules
+    self.xt = xt
+    self.root = self.xt.root
+    self.__dict__.update(sent.__dict__)
+
+    # Add some additional useful attributes
+    self.tagged_sent = ' '.join(tag_seqs(self.words, self.idxs, self.labels))
+
+  def render(self):
+    self.xt.render_tree(self.idxs)
+
+  def __repr__(self):
+    return '<Relation: {} - {}>'.format(self.e1_idxs, self.e2_idxs)
+
+class Relations(Extractions):
+  def __init__(self, e1, e2, sents):
+    self.e1 = e1
+    self.e2 = e2
+    super(Relations, self).__init__(sents)
+    self.relations = self.extractions
+
+  def _apply(self, sent):
+    xt = corenlp_to_xmltree(sent)
+    for e1_idxs in self.e1.apply(sent):
+      for e2_idxs in self.e2.apply(sent):
+        yield Relation(e1_idxs, e2_idxs, self.e1.label, self.e2.label, sent, xt)
+  
+  def _get_features(self, method='treedlib'):
+    get_feats = compile_relation_feature_generator()
+    f_index = defaultdict(list)
+    for j,ext in enumerate(self.extractions):
+      for feat in get_feats(ext.root, ext.e1_idxs, ext.e2_idxs):
+        f_index[feat].append(j)
+    return f_index
+
+class Entity:
+  def __init__(self, idxs, name, sent, xt):
+    self.name = name
+    self.idxs = idxs
+    # Absorb XMLTree and Sentence object attributes for access by rules
+    self.xt = xt
+    self.root = self.xt.root
+    self.__dict__.update(sent.__dict__)
+
+    # Add some additional useful attributes
+    self.tagged_sent = ' '.join(tag_seqs(self.words, [self.idxs], 'ENT'))
+
+  def render(self):
+    self.xt.render_tree([self.idxs])
+
+  def __repr__(self):
+    return '<Entity: {}>'.format(self.name)
+
+class Entities(Extractions):
+  def __init__(self, sents):
+    super(Entities, self).__init__(sents)
+    self.entities = self.extractions
+
+  def _apply(self, sent):
+    xt = corenlp_to_xmltree(sent)
+    # Search for consecutive named entity blocks
+    nr_pos = ['NN', 'NNS', 'NNP', 'NNPS']
```

Comment:
@henryre I think I quickly responded over email agreeing with this choice, but I actually don't think it should be hardcoded- examples have already come up where we want other POS types, or more flexibility in general